topic.go

```diff
@@ -347,9 +336,7 @@ func WithSecretKeyAndPeerId(key crypto.PrivKey, pid peer.ID) PubOpt {
 // Close closes down the topic. Will return an error unless there are no active event handlers or subscriptions.
 // Does not error if the topic is already closed.
 func (t *Topic) Close() error {
-	t.mux.Lock()
-	defer t.mux.Unlock()
-	if t.closed {
+	if t.closed.Load() {
```

Comment:
There is a race here: you want to do `Swap` here and if old result was `false` then proceed with closing.

Otherwise 2 goroutines can hit Close, load false and proceed.